micropython/drivers/imu/bmm150/bmm150.py

```diff
@@ -0,0 +1,188 @@
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+
+Basic example usage:
+
+import time
+from bmm150 import BMM150
+from machine import Pin, SPI, I2C
+
+# Init in I2C mode.
+imu = BMM150(I2C(1, scl=Pin(15), sda=Pin(14)))
+
+# Or init in SPI mode.
+# TODO: Not supported yet.
+# imu = BMM150(SPI(5), cs_pin=Pin(10))
+
+while (True):
+    print('magnetometer: x:{:>8.3f} y:{:>8.3f} z:{:>8.3f}'.format(*imu.read_magnet()))
+    time.sleep_ms(100)
+"""
+
+import array
+import time
+from micropython import const
+
+_DEFAULT_ADDR = const(0x10)
+_CHIP_ID = const(0x40)
+_DATA = const(0x42)
+_POWER = const(0x4B)
+_OPMODE = const(0x4C)
+_INT_STATUS = const(0x4A)
+_TRIM_X1 = const(0x5D)
+_TRIM_Y1 = const(0x5E)
+_TRIM_Z4_LSB = const(0x62)
+_TRIM_Z2_LSB = const(0x68)
+_XYAXES_FLIP = const(-4096)
+_ZHAXES_FLIP = const(-16384)
+
+
+class BMM150:
+    def __init__(
+        self,
+        bus,
+        cs_pin=None,
+        address=_DEFAULT_ADDR,
+        magnet_odr=30,
+    ):
+        """Initalizes the Magnetometer.
+        bus: IMU bus
+        address: I2C address (in I2C mode).
+        cs_pin: SPI CS pin (in SPI mode).
+        magnet_odr: (2, 6, 8, 10, 15, 20, 25, 30)
+        """
+        self.bus = bus
+        self.cs_pin = cs_pin
+        self.address = address
+        self._use_i2c = hasattr(self.bus, "readfrom_mem")
+
+        ODR = (10, 2, 6, 8, 15, 20, 25, 30)
+
+        # Sanity checks
+        if not self._use_i2c:
+            raise ValueError("SPI mode is not supported")
+        if not magnet_odr in ODR:
+            raise ValueError("Invalid sampling rate: %d" % magnet_odr)
+
+        # Perform soft reset, and power on.
+        self.__write_reg(_POWER, 0x83)
+        time.sleep_ms(10)
+
+        if self.__read_reg(_CHIP_ID) != 0x32:
+            raise OSError("No BMM150 device was found at address 0x%x" % (self.address))
+
+        # Configure the device.
+        # ODR | OP: Normal mode
+        self.__write_reg(_OPMODE, ODR.index(magnet_odr) << 3)
+
+        # Read trim registers.
+        trim_x1y1 = self.__read_reg(_TRIM_X1, 2)
+        trim_xyz_data = self.__read_reg(_TRIM_Z4_LSB, 4)
+        trim_xy1xy2 = self.__read_reg(_TRIM_Z2_LSB, 10)
+
+        self.trim_x1 = trim_x1y1[0]
+        self.trim_y1 = trim_x1y1[1]
+        self.trim_x2 = trim_xyz_data[2]
+        self.trim_y2 = trim_xyz_data[3]
+        self.trim_z1 = (trim_xy1xy2[3] << 8) | trim_xy1xy2[2]
+        self.trim_z2 = (trim_xy1xy2[1] << 8) | trim_xy1xy2[0]
+        self.trim_z3 = (trim_xy1xy2[7] << 8) | trim_xy1xy2[6]
+        self.trim_z4 = (trim_xyz_data[1] << 8) | trim_xyz_data[0]
+        self.trim_xy1 = trim_xy1xy2[9]
+        self.trim_xy2 = trim_xy1xy2[8]
+        self.trim_xyz1 = ((trim_xy1xy2[5] & 0x7F) << 8) | trim_xy1xy2[4]
+
+        # Allocate scratch buffer.
+        self.scratch = memoryview(array.array("h", [0, 0, 0, 0]))
+
+    def __read_reg(self, reg, size=1):
```

Comment:
Was the intention to change these method names to start with a single underscore?

Reply:
Yes missed this one.